api/views.py

```diff
@@ -0,0 +1,29 @@
+from flask import Blueprint, jsonify
+from .models import BookSchema
+from .mock_db import db
+
+
+def jsonify_schema_dump(*args, **kwargs):
+    """Wrapper around jsonify that sets the Content-Type of the response to
+    application/vnd.api+json.
+    """
+    response = jsonify(*args, **kwargs)
+    response.mimetype = 'application/vnd.api+json'
+    return response
+
+
+books_bp = Blueprint('books', __name__)
+
+
+@books_bp.route('/')
+def get_all():
+    books = db['books']
+    data = BookSchema(many=True).dump(books)
+    return jsonify_schema_dump(data)
+
+
+@books_bp.route('/<int:book_id>')
+def get_by_id(book_id):
```

Comment:
The responses don't look right

```json
[
    {
        "data": {
            "attributes": {
                "date_published": "1990-06-12",
                "isbn": "0-3605-3454-6",
                "title": "Test title 1"
            },
            "id": "1",
            "links": {
                "self": "/books/1"
            },
            "type": "books"
        },
        "links": {
            "self": "/books/1"
        }
    },
    {}
]
```

Notice that for a single resource an array is returned and also there is a empty {} hash at the end.